url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/Intervention.java

```diff
@@ -0,0 +1,229 @@
+
+import javax.jdo.annotations.NotPersistent;
+import javax.jdo.annotations.PersistenceCapable;
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.sign.AbstractModel;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class Intervention extends AbstractModel {
+
+	private List<Vehicule> vehicules;
+	private List<Cible> cibles;
+	private List<Source> sources;
+	private List<Action> actions;
+	private List<Message> messages;
+	private List<Implique> impliques;
+	private List<DemandeMoyen> demandes_moyen;
+
+
+	public Intervention() {
+		super(null, null, new Position(0, 0));
+		this.vehicules = new ArrayList<Vehicule>();
+		this.cibles = new ArrayList<Cible>();
+		this.sources = new ArrayList<Source>();
+		this.actions = new ArrayList<Action>();
+		this.messages = new ArrayList<Message>();
+		this.impliques = new ArrayList<Implique>();
+		this.demandes_moyen = new ArrayList<DemandeMoyen>();
+	}
+
+	public Intervention(String uid) {
+		super(uid, null, new Position(0, 0));
+		this.vehicules = new ArrayList<Vehicule>();
+		this.cibles = new ArrayList<Cible>();
+		this.sources = new ArrayList<Source>();
+		this.actions = new ArrayList<Action>();
+		this.messages = new ArrayList<Message>();
+		this.impliques = new ArrayList<Implique>();
+		this.demandes_moyen = new ArrayList<DemandeMoyen>();
+	}
+
+	public Intervention(JSONObject json) throws InvalidJSONException {
+		super(json);
+
+		this.vehicules = new ArrayList<Vehicule>();
+		this.cibles = new ArrayList<Cible>();
+		this.sources = new ArrayList<Source>();
+		this.actions = new ArrayList<Action>();
+		this.messages = new ArrayList<Message>();
+		this.impliques = new ArrayList<Implique>();
+		this.demandes_moyen = new ArrayList<DemandeMoyen>();
+
+		try {
+			JSONArray jsar = json.getJSONArray("vehicules");
+			for (int i = 0; i < jsar.length(); i++) {
+				vehicules.add(new Vehicule(jsar.getJSONObject(i)));
+			}
+
+			jsar = json.getJSONArray("cibles");
+			for (int i = 0; i < jsar.length(); i++) {
+				cibles.add(new Cible(jsar.getJSONObject(i)));
+			}
+
+			jsar = json.getJSONArray("sources");
+			for (int i = 0; i < jsar.length(); i++) {
+				sources.add(new Source(jsar.getJSONObject(i)));
+			}
+
+			jsar = json.getJSONArray("actions");
+			for (int i = 0; i < jsar.length(); i++) {
+				actions.add(new Action(jsar.getJSONObject(i)));
+			}
+
+			jsar = json.getJSONArray("messages");
+			for (int i = 0; i < jsar.length(); i++) {
+				messages.add(new Message(jsar.getJSONObject(i)));
+			}
+
+			jsar = json.getJSONArray("impliques");
+			for (int i = 0; i < jsar.length(); i++) {
+				impliques.add(new Implique(jsar.getJSONObject(i)));
+			}
+			
+			jsar = json.getJSONArray("demandes_moyen");
+			for (int i = 0; i < jsar.length(); i++) {
+				demandes_moyen.add(new DemandeMoyen(jsar.getJSONObject(i)));
+			}
+		} catch (JSONException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)